mercury-framework/mercury-grpc-service/src/main/java/com/redhat/mercury/GrpcService.java

```diff
@@ -0,0 +1,123 @@
+
+        for (FuntionalPattern fp : FuntionalPatterns) {
+            data += String.format("import com.redhat.mercury.%s.%s.api.%s.%s;\n",
+                                  sdNameCamel.toLowerCase(), version, fp.getFirstCap().toLowerCase(), fp.getFirstCap());
+        }
+
+        data += String.format("\n" +
+                              "import io.quarkus.grpc.GrpcClient;\n" +
+                              "\n" +
+                              "@ApplicationScoped\n" +
+                              "public class %sClient {\n" +
+                              "\n",
+                              sdNameCamel);
+
+        for (FuntionalPattern fp : FuntionalPatterns) {
+            data += String.format("    @GrpcClient(\"%s\")\n" +
+                                  "    %s %s;\n" +
+                                  "\n",
+                                  fp.getWithSD(), fp.getFirstCap(), fp.getFirstNotCap());
+        }
+
+        for (FuntionalPattern fp : FuntionalPatterns) {
+            data += String.format("    public %s %s() {\n" +
+                                  "        return %s;\n" +
+                                  "    }\n" +
+                                  "\n",
+                                  fp.getFirstCap(), fp.getWithGet(), fp.getFirstNotCap());
+        }
+
+        data += "}";
+
+        // Create service domain client file in target directory
+        File clientDir = new File(String.format("service-domains/%s/%s-client/target/classes/com/redhat/mercury/%s/%s/client",
+                                                sdName, sdName, sdNameCamel.toLowerCase(), version));
+        File clientFile = new File(clientDir, sdNameCamel+"Client.java");
+        try (BufferedWriter clientBw = new BufferedWriter(new FileWriter(clientFile,false))) {
+            clientBw.write(data);
+        } catch(IOException e){
+            LOGGER.error("context", e);
+        }
+
+        // Create service domain application.properties file content
+        String properties = "";
+        for (FuntionalPattern fp : FuntionalPatterns) {
+            properties += String.format("quarkus.grpc.clients.%s.host=${mercury.%s.service.host:%s}\n" +
+                            "quarkus.grpc.clients.%s.port=${mercury.%s.service.port:%s}\n",
+                    fp.getWithSD(), sdName, sdName, fp.getWithSD(), sdName, port);
+        }
+
+        // Create service domain application.properties file in target directory
+        File propertiesDir = new File(String.format("service-domains/%s/%s-client/target/classes",
+                                                    sdName, sdName));
+        File propertiesFile = new File(propertiesDir, "application.properties");
+        try (BufferedWriter propertiesBw = new BufferedWriter(new FileWriter(propertiesFile,false))) {
+            propertiesBw.write(properties);
+        } catch(IOException e){
+            LOGGER.error("context", e);
```

Comment:
I think that if an error happens the process should terminate.
Also add a more descriptive message for the error